tests/contrib/test_pydantic.py

```diff
@@ -0,0 +1,346 @@
+import dataclasses
+import uuid
+from datetime import date, datetime, timedelta
+from ipaddress import IPv4Address
+from typing import Annotated, Any, List, Sequence, Tuple, TypeVar, Union
+
+from annotated_types import Len
+from pydantic import BaseModel, Field, WithJsonSchema
+
+from temporalio import activity, workflow
+from temporalio.client import Client
+from temporalio.contrib.pydantic import pydantic_data_converter
+from temporalio.worker import Worker
+
+SequenceType = TypeVar("SequenceType", bound=Sequence[Any])
+ShortSequence = Annotated[SequenceType, Len(max_length=2)]
+
+
+class MyPydanticModel(BaseModel):
+    ip_field: IPv4Address
+    string_field_assigned_field: str = Field()
+    string_field_with_default: str = Field(default_factory=lambda: "my-string")
+    annotated_list_of_str: Annotated[
+        List[str], Field(), WithJsonSchema({"extra": "data"})
+    ]
+    str_short_sequence: ShortSequence[List[str]]
+
+    def _check_instance(self):
+        assert isinstance(self.ip_field, IPv4Address)
+        assert isinstance(self.string_field_assigned_field, str)
+        assert isinstance(self.string_field_with_default, str)
+        assert isinstance(self.annotated_list_of_str, list)
+        assert isinstance(self.str_short_sequence, list)
+        assert self.annotated_list_of_str == ["my-string-1", "my-string-2"]
+        assert self.str_short_sequence == ["my-string-1", "my-string-2"]
+
+
+class MyPydanticDatetimeModel(BaseModel):
```

Comment:
Just to make us feel better, can we toss timedelta in here or somewhere and make sure it works (I'm sure it does since we don't proxy it)

Reply:
Yes, was just doing that in fact! 2ff63ce66b9e489efde742a611fba5439b829edf